src/main.rs

```diff
@@ -45,6 +52,58 @@ impl<W: std::io::Write> RockPaperScissorsGame<W>{
       _ => ShapesInput::Unrecognized
     }
   }
+
+  fn play_round(&mut self) -> Result<bool, std::io::Error> {
+    let input = self.evaluate_user_input();
+    dbg!(input);
+    match input {
+      ShapesInput::Unrecognized => {
+        writeln!(
+          self.writer,
+          "Input cannot be parsed as a a rock / paper / scissors, try again!"
+        )?;
+        Ok(true)
+      },
+      shapePlayed => {
+        let result = self.evaluateAsLeft(shapePlayed, self.computer_shape);
+        dbg!(
+          // "you played {}, they played {}, result = {}",
+          shapePlayed, self.computer_shape, result
+        );
+        Ok(false)
+      }
+    }
+  }
+
+  fn transformShapeToInt(&mut self, shape: ShapesInput) -> i32 {
+    match shape {
+      ShapesInput::Rock => 0,
+      ShapesInput::Paper => 1,
+      ShapesInput::Scissors => 2,
+      _ => 9999
```

Comment:
tento case by ti mal pomoct si uvedomit, ze to Unrecognized asi nebol dobry napad :D